From 6c7c21e9248379d7283bd500414b2f627d9747f3 Mon Sep 17 00:00:00 2001
From: Frank Wunderlich <frank-w@public-files.de>
Date: Fri, 17 Oct 2025 17:42:17 +0200
Subject: [PATCH 101/116] net: phy: mtk-2p5ge: fix crash because of missing
 callbacks

commit
5e3aae2d3271 2025-08-27 net: phy: mtk-2p5ge: Add LED support for MT7988

introduces new callbacks which crash kernel if missing

Unable to handle kernel execute from non-executable memory at virtual address 0000000000000000
---
 drivers/net/phy/mediatek/mtk-2p5ge.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/net/phy/mediatek/mtk-2p5ge.c b/drivers/net/phy/mediatek/mtk-2p5ge.c
index b4508bf7de67..bf4e2d6a55fa 100644
--- a/drivers/net/phy/mediatek/mtk-2p5ge.c
+++ b/drivers/net/phy/mediatek/mtk-2p5ge.c
@@ -724,6 +724,11 @@ static struct phy_driver mtk_2p5gephy_driver[] = {
 		.resume = genphy_resume,
 		.read_page = mtk_phy_read_page,
 		.write_page = mtk_phy_write_page,
+		.led_blink_set = mt798x_2p5ge_phy_led_blink_set,
+		.led_brightness_set = mt798x_2p5ge_phy_led_brightness_set,
+		.led_hw_is_supported = mt798x_2p5ge_phy_led_hw_is_supported,
+		.led_hw_control_get = mt798x_2p5ge_phy_led_hw_control_get,
+		.led_hw_control_set = mt798x_2p5ge_phy_led_hw_control_set,
 	},
 	{
 		PHY_ID_MATCH_MODEL(MTK_2P5GPHY_ID_MT7988),
-- 
2.39.5

